Makefile

```diff
@@ -426,9 +426,15 @@ coverage/:
 	mkdir -p $@
 
 .PHONY: generate
-generate:
+generate: generate-tools
 	${GO} generate -gcflags '-N -l' -ldflags "-X main.VERSION=$(TAG)" ./...
-	echo "[OK] Files added to pipeline template directory!"
+	@echo "[OK] Files added to pipeline template directory!"
+
+.PHONY: generate-tools
+generate-tools:
+	@which mockgen > /dev/null || echo "Installing 'mockgen'" && ${GO} install github.com/golang/mock/mockgen@v1.6.0
```

Comment:
Looks great. Though mockgen v1.6.0 doesn't support generics yet and will fail when trying to mock interfaces with generics. Installing `github.com/golang/mock/mockgen@v1.7.0-rc.1` worked for me